src/neural_data_simulator/decoder/run_decoder.py

```diff
@@ -65,14 +81,18 @@ def _read_decode_send(
 def run():
     """Run the decoder loop."""
     initialize_logger(SCRIPT_NAME)
-
-    settings = cast(
+    args = _parse_args()
+    settings: _Settings = cast(
         _Settings,
-        get_script_settings(
-            _parse_args_settings_path(), "settings_decoder.yaml", _Settings
+        load_settings(
+            args.settings_path,
+            settings_parser=_Settings,
+            override_dotlist=args.overrides,
         ),
     )
+
     configure_logger(SCRIPT_NAME, settings.log_level)
+    logger.debug(f"run_kecoder settings:\n{yaml.dump(settings.dict())}")
```

Comment:
```suggestion
    logger.debug(f"run_decoder settings:\n{yaml.dump(settings.dict())}")
```

Reply:
Do you think it would make sense to have an arg to print the settings instead?

something like: `decoder --print-settings` that only prints it.

Reply:
A `--print-settings` option would be good! I'll add that.

I'd still like to keep the `logger.debug` here as well. It won't be printed by default, but it would be helpful for an experimenter taking a closer-look (i.e. they've specified DEBUG logging) to have the exact settings collated with their .logs

Reply:
Added the `--print-settings-only`/`-p` option.

```
❯ streamer -o log_level=DEBUG streamer.lsl_chunk_frequency=50 --print-settings-only                              
_Settings(
│   log_level=<LogLevel._DEBUG: 'DEBUG'>,
│   streamer=Streamer(
│   │   blackrock=Blackrock(
│   │   │   output=Output(lsl=LSLSimplifiedOutputModel(channel_format=<LSLChannelFormatType._INT16: 'int16'>, instrument=_Instrument(manufacturer='Blackrock Neurotech', model='Playback', id=1))),
│   │   │   input=Input(file=PosixPath('sample_data/blackrock-file.ns6'))
│   │   ),
│   │   npz=NPZ(
│   │   │   output=Output(
│   │   │   │   sampling_rate=50.0,
│   │   │   │   n_channels=2,
│   │   │   │   lsl=LSLOutputModel(
│   │   │   │   │   channel_format=<LSLChannelFormatType._FLOAT32: 'float32'>,
│   │   │   │   │   stream_name='NDS-Behavior',
│   │   │   │   │   stream_type='Behavior',
│   │   │   │   │   source_id='SimulatedBehavior',
│   │   │   │   │   instrument=_Instrument(manufacturer='Blackrock Neurotech', model='Simulated', id=0),
│   │   │   │   │   channel_labels=None
│   │   │   │   )
│   │   │   ),
│   │   │   input=Input(file=PosixPath('sample_data/session_4_behavior_standardized.npz'), timestamps_array_name='timestamps_train', data_array_name='vel_train')
│   │   ),
│   │   input_type=<StreamerInputType.NPZ: 'npz'>,
│   │   lsl_chunk_frequency=50.0,
│   │   stream_indefinitely=True
│   )
)
```

With more readable highlighting (via [rich-printing](https://rich.readthedocs.io/en/stable/introduction.html)) in the terminal:
<img width="540" alt="image" src="https://github.com/agencyenterprise/neural-data-simulator/assets/3221512/89b3ecb2-844e-4d9a-9e0d-ae260c198612">

Reply:
I didn't add this option to `run_closed_loop.py`, as it would have made the `control-file` logic messier, and it's available to users to run the scripts separately and the args are now logged via `_run_process`, e.g.:
```
INFO [nds-run-closed-loop]: Starting process: ['encoder', '--overrides', 'log_level=DEBUG', 'streamer.lsl_chunk_frequency=50']
```